include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -658,9 +671,9 @@ distribute_local_to_global (const Vector<LocalType>      &local_vector,
         if (val != 0)
           for (size_type j=0; j<n_local_dofs; ++j)
             {
-              if (is_constrained(local_dof_indices[j]) == false)
+              if (is_constrained(local_dof_indices_row[j]) == false)
                 {
-                  global_vector(local_dof_indices[j]) -= val * local_matrix(j,i);
+                  global_vector(local_dof_indices_row[j]) -= val * local_matrix(j,i);
```

Comment:
This entry takes an unconstrained row (correctly identified by local_dof_indices_row) and multiplies the known values of the solution vector (identified by the correct index through 'col').